buildman/manager/ephemeral.py

```diff
@@ -41,271 +58,114 @@
-
-        # Create the build information block for the registered realm.
-        build_job = BuildJob(AttrDict(realm_spec["job_queue_item"]))
-        execution_id = realm_spec.get("execution_id", None)
-        executor_name = realm_spec.get("executor_name", "EC2Executor")
-
-        logger.debug("Registering realm %s with manager: %s", realm_spec["realm"], realm_spec)
-        component = self.register_component(
-            realm_spec["realm"], BuildComponent, token=realm_spec["token"]
-        )
-
-        build_info = BuildInfo(
-            component=component,
-            build_job=build_job,
-            execution_id=execution_id,
-            executor_name=executor_name,
-        )
-
-        self._component_to_job[component] = build_job
-        self._build_uuid_to_info[build_job.build_uuid] = build_info
-
-        logger.debug("Registered realm %s with manager", realm_spec["realm"])
-        return component
-
-    @property
-    def registered_executors(self):
-        return self._ordered_executors
-
-    async def _register_existing_realms(self):
-        try:
-            all_realms = await self._orchestrator.get_prefixed_keys(self._realm_prefix)
-
-            # Register all existing realms found.
-            encountered = {
-                self._register_realm(json.loads(realm_data)) for _realm, realm_data in all_realms
-            }
-
-            # Remove any components not encountered so we can clean up.
-            for component, job in iteritems(self._component_to_job):
-                if not component in encountered:
-                    self._component_to_job.pop(component, None)
-                    self._build_uuid_to_info.pop(job.build_uuid, None)
+        logger.debug("calling orchestrator_from_config")
+        self._orchestrator = orchestrator_from_config(manager_config)
 
-        except KeyError:
-            pass
+        logger.debug("setting on_key_change callbacks for job expiry, cancel")
+        self._orchestrator.on_key_change(self._job_prefix, self._job_expired_callback)
+        self._orchestrator.on_key_change(self._cancel_prefix, self._job_cancelled_callback)
 
     def _load_executor(self, executor_kind_name, executor_config):
         executor_klass = EphemeralBuilderManager.EXECUTORS.get(executor_kind_name)
         if executor_klass is None:
             logger.error("Unknown executor %s; skipping install", executor_kind_name)
             return
 
-        executor = executor_klass(executor_config, self.manager_hostname)
+        executor = executor_klass(executor_config, self._manager_hostname)
         if executor.name in self._executor_name_to_executor:
             raise Exception("Executor with name %s already registered" % executor.name)
 
         self._ordered_executors.append(executor)
         self._executor_name_to_executor[executor.name] = executor
 
+    def generate_build_token(self, token_type, build_id, job_id, expiration):
```

Comment:
This just seems like a very thin wrapper around `build_token()`. Why not call it directly? Same with `verify_build_token()`.

Reply:
This just saves us from having to explicitly pass the hostname and instance keys. But I can call it directly if you want.

Reply:
It's not very important. Sometimes our code is a little hard to follow since we have a ton of abstraction layers. Just hoping to reduce that over time. I'll go ahead and resolve this one.

Thanks!